test/go-tests/test_selfhealing.go

```diff
@@ -3,14 +3,103 @@ package go_tests
 import (
 	"fmt"
 	"github.com/keptn/go-utils/pkg/api/models"
-	"github.com/keptn/go-utils/pkg/common/osutils"
 	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
 	"github.com/stretchr/testify/require"
 	"os"
+	"strings"
 	"testing"
 	"time"
 )
 
+const unleashServiceK8sManifest = `---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: unleash-service
+spec:
+  selector:
+    matchLabels:
+      run: unleash-service
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: unleash-service
+        app.kubernetes.io/name: unleash-service
+        app.kubernetes.io/version: 0.3.2
+    spec:
+      serviceAccountName: keptn-default
+      containers:
+        - name: unleash-service
+          image: keptncontrib/unleash-service:0.3.2
+          ports:
+            - containerPort: 8080
+          env:
+            - name: EVENTBROKER
+              value: 'http://localhost:8081/event'
+            - name: CONFIGURATION_SERVICE
+              value: 'http://configuration-service:8080'
+          envFrom:
+            - secretRef:
+                name: unleash
+                optional: true
+        - name: distributor
+          image: ${distributor-image}
+          ports:
+            - containerPort: 8080
+          resources:
+            requests:
+              memory: "16Mi"
+              cpu: "25m"
+            limits:
+              memory: "32Mi"
+              cpu: "250m"
+          env:
+            - name: PUBSUB_URL
+              value: 'nats://keptn-nats'
+            - name: PUBSUB_TOPIC
+              value: 'sh.keptn.>'
+            - name: PUBSUB_RECIPIENT
+              value: '127.0.0.1'
+            - name: VERSION
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: 'metadata.labels[''app.kubernetes.io/version'']'
+            - name: K8S_DEPLOYMENT_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: 'metadata.labels[''app.kubernetes.io/name'']'
+            - name: K8S_POD_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.name
+            - name: K8S_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.namespace
+            - name: K8S_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: unleash-service
+  labels:
+    run: unleash-service
+spec:
+  ports:
+    - port: 8080
+      protocol: TCP
+  selector:
+    run: unleash-service`
```

Comment:
This would be fitting for a yml files under te `go-tests/assets` folder

Reply:
as discussed, I will create an issue for refactoring this integration test to not be dependent on the unleash service anymore